### PR TITLE
Mandatory fields

### PR DESCRIPTION
**Pull request by Julias0**

### CODE REVIEW

**Comment by Julias0** [comment]
Screenshot 2021-07-22 at 4 56 04 PM
Screenshot 2021-07-22 at 4 56 10 PM
Screenshot 2021-07-22 at 4 56 24 PM
Screenshot 2021-07-22 at 4 47 16 PM
Screenshot 2021-07-22 at 4 47 33 PM
Screenshot 2021-07-22 at 4 47 40 PM
Screenshot 2021-07-22 at 4 55 24 PM
Screenshot 2021-07-22 at 4 55 31 PM
Screenshot 2021-07-22 at 4 55 50 PM

**Comment by Julias0** [comment]
Mileage after Mandatory Fields Changes
Screenshot 2021-07-26 at 1 18 16 PM
Screenshot 2021-07-26 at 1 18 28 PM
Screenshot 2021-07-26 at 1 19 15 PM

Screenshot 2021-07-26 at 1 18 08 PM

**Comment by Julias0** [comment]
Add Expense after Mandatory Fields Changes
Screenshot 2021-07-26 at 1 19 46 PM
Add
Screenshot 2021-07-26 at 1 19 57 PM
Screenshot 2021-07-26 at 1 20 05 PM

Screenshot 2021-07-26 at 1 19 37 PM

**Comment by Julias0** [comment]
Per Diem after Mandatory Field Changes
Screenshot 2021-07-26 at 1 20 34 PM
Screenshot 2021-07-26 at 1 20 39 PM
Screenshot 2021-07-26 at 1 20 48 PM

File: src/app/fyle/add-edit-expense/add-edit-expense.page.scss [hunk 1]
```diff
@@ -260,10 +266,20 @@
...
     background: #fff;
     line-height: 1;
-    margin-bottom: 12px;
+    &__margin {
+      margin-right: 12px;    
+    }
+  }
+
+  &--expansion-btn {
+    @extend %expansion-btn;
```
Comment (abhishek1234321) [inline src/app/fyle/add-edit-expense/add-edit-expense.page.scss:278]: this is not the best way no? meaning we aren't reusing SCSS here

File: src/app/fyle/add-edit-expense/add-edit-expense.page.ts [hunk 2]
```diff
@@ -1900,6 +1901,8 @@ export class AddEditExpensePage implements OnInit {
 
     this.mode = this.activatedRoute.snapshot.params.id ? 'edit' : 'add';
 
+    this.isExpandedView = !(this.mode === 'add');
```
Comment (abhishek1234321) [inline src/app/fyle/add-edit-expense/add-edit-expense.page.ts:1904]: `this.mode !== 'add'` easier on the mind!

File: src/app/fyle/add-edit-expense/fy-currency/fy-currency.component.html [hunk 3]
```diff
@@ -20,7 +20,7 @@
     </div>
   </form>
 </div>
-<ng-container *ngIf="value?.currency && value?.orig_currency">
+<ng-container *ngIf="value?.currency && value?.orig_currency && exchangeRate">
```
Comment (abhishek1234321) [inline src/app/fyle/add-edit-expense/fy-currency/fy-currency.component.html:23]: why is this needed?

File: src/app/fyle/add-edit-mileage/add-edit-mileage.page.scss [hunk 4]
```diff
@@ -446,4 +450,17 @@
   &--location {
     flex-grow: 1;
   }
+
+  &--grid {
+    --ion-grid-padding: 0;
+    --ion-grid-column-padding: 0;
+  }
+
+  &--expansion-btn {
```
Comment (abhishek1234321) [inline src/app/fyle/add-edit-mileage/add-edit-mileage.page.scss:459]: same as above

**Review by abhishek1234321 — APPROVED** [review]

File: src/app/fyle/add-edit-expense/fy-currency/fy-currency.component.html (hunk 3 above)
Comment (Julias0) [reply src/app/fyle/add-edit-expense/fy-currency/fy-currency.component.html:23]: Resolved offline
